src/formatDate.js

```diff
@@ -50,7 +50,63 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.pop();
+  const toSeparator = toFormat.pop();
```

Comment:
Don't mutate incoming arrays.

Reply:
Thank you. I have made changes. I don't understand why it's failing to work with something like this  
```
const fromSeparator = fromFormat[3];
 const toSeparator = toFormat[3];
```

Please help.

Reply:
Well, I suppose that you iterate through full length of the array. It reaches the last element and your `switch/case` structure throws an error.